### ISSUE TITLE
Widget and job package manager

### ISSUE THREAD

**Issue by davejlong** [kittoframework/kitto#18]
Building a package manager to install widgets and jobs would help the process of building a dashboard and get people from idea to app even faster.

The idea is to that someone can create a package which includes both the widget javascript and stylesheet as a Gist or some other package form. Then we could add a Mix task for installing the widget with:

```
mix kitto.install.widget davejlong/4990174 # <= davejlong/4990174 being a Gist that includes the widget
```

Creating a job would be just as simple:

```
mix kitto.install.job davejlong/4990167 # <= davejlong/4990167 being a Gist that includes the job file
```

**Comment by zorbash** [comment]
I totally agree we should make it easier for the user to quickly populate dashboards with ready-made widgets/jobs.

Dashing already does something similar to what you propose (see: https://github.com/Shopify/dashing/blob/master/lib/dashing/cli.rb#L92).

Why do you suggest to have different tasks for widget, job installation?

**Comment by davejlong** [comment]
I suppose there's no need to have different tasks. In theory the implementation would just be to pull a Gist. Group sass and js files into a widget and exs files into a job. If neither a widget, nor a job are found, fail with an exception saying as such.

**Comment by zorbash** [comment]
Dashing plugin system works fine for widgets / jobs without external dependencies.
If a widget needs an npm package to work, the user will still have to do some manual work to tweak her package.json and do `npm install`.

**Comment by zorbash** [comment]
I'm doing some research on how we could incorporate a mechanism like http://docs.operable.io/docs/bundle-configs so that we can have a registry for to showcase their dashboards/widgets and a simple config for someone to declare kitto plugin dependencies.

**Comment by davejlong** [comment]
I was going to start doing some research tonight on building something out
a little more advanced than Dashing as well. I think having a web based
search engine for finding packages will be important to the growth of Kitto
and something that was always lacking in Dashing. Perhaps we can have a
further conversation on this in Slack?

On November 16, 2016 at 1:55:12 PM, Dimitrios Zorbas (
notifications@github.com) wrote:

I'm doing some research on how we could incorporate a mechanism like
http://docs.operable.io/docs/bundle-configs so that we can have a registry
for to showcase their dashboards/widgets and a simple config for someone to
declare kitto plugin dependencies.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
https://github.com/kittoframework/kitto/issues/18#issuecomment-261037201,
or mute the thread

**Comment by zorbash** [comment]
Meanwhile I added a script to help updating the core widgets.

```
sh -c "$(curl -fsSL https://gist.githubusercontent.com/Zorbash/98afb51cb7287ff7fe3b2b9a3331dc5f/raw/16ec3aa6e7ef88ee121ee4c30fc982e5f8775850/kitto_updater.sh)"
```
